### PR TITLE
exterminator midround minor antag

### PR DESCRIPTION
**Pull request by deltanedas**
## About the PR
- added a station event to spawn a terminator for a random player
- added an admin smite to spawn a terminator targeting a specific naughty player

design doc: https://docs.spacestation14.com/en/proposals/deltanedas-exterminator.html

its a fully terminator themed midround minor antag that only has to kill 1 player
similar to chrono legionnaires in tg except:
- no teleporting and shit
- not admeme exclusive, it can happen to anyone
- no epic modsuit or anything you are sent back in time naked so better rob a tider quick
- highly resistant to most damage and strong punches, with a weakness to fire
- once you take 100 burn damage or 200 of any damage your flesh exterior will fall off and you will be a relatively weak endoskeleton.
  the endoskeleton has stronger melee but has no hands or anything and is much slower
  it's intended as a last resort to try finish off your target
- after you rob a gun store and shoot up your target at technoir, your objectives are complete and you can either double gib yourself in an emagged recycler or find a bang stick [gibber] to "shut down".

TODO:
- [X] finish recoloring skeleton real
- [X] use codersprite
- [X] fix funny ghost role spawner retake bug thing
- [X] remove lung
- [X] use non codersprite

inb4 muh 2029 vs 2125

i am now a lawyer :trollface::trollface::trollface:

depends on #20186

## Why / Balance
nobody said no on the discussion :trollface: and its a nice midround that is something security/chem/sci may have to fend off, but not a roundender ~~unless its sosa~~
like more powerful evil twin from nyano without being a twin

## Technical details
- added `PopupBehavior` for its special endoskeleton gibbing/burning transformations to use
- terminator rule has a `CreateSpawner` function which the smite uses
- the body stuff is very sus

## Media
codersprite is in

objective and briefing

text is green

endoskeleton ops

- [ ] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes
probably not

**Changelog**
:cl:
- add: Cybersun in the future is sending NT-800 infiltrators back to your time to exterminate people on the station that it deems important to the future.

### CODE REVIEW

**Comment by noverd** [comment]
why not lol

File: Content.Server/GameTicking/Rules/TerminatorRuleSystem.cs
Comment (metalgearsloth) [inline Content.Server/GameTicking/Rules/TerminatorRuleSystem.cs:43]: Needs to support multiple gamerules on a target, just make each one a terminator prolly.

**Comment by HoidC** [comment]
yeah so I know its the terminator but taking 10 percent damage on top of armour being equippable is going to make it literally unkillable

File: Content.Server/GameTicking/Rules/TerminatorRuleSystem.cs
Comment (deltanedas) [reply Content.Server/GameTicking/Rules/TerminatorRuleSystem.cs:43]: then objectives wont be grouped and if theres 2 terminators that targeted same person (still extremely low odds but possible) itll say
there was 1 t-800
abc kill bob

there was 1 t-800
xyz kill bob

**Comment by deltanedas** [comment]
watch the film pro they shoot him with shotguns and assault rifles and does nothing
oil tanker? stripped the endoskeleton clean :trollface:

**Comment by HoidC** [comment]
> watch the film pro they shoot him with shotguns and assault rifles and does nothing
> oil tanker? stripped the endoskeleton clean :trollface:

doesn’t the terminator also take 0.1 heat damage

**Comment by HoidC** [comment]
nvm I assume they take full explosion damage 

encouraging the crew to suicide bomb the valid is kinda iffy though

**Comment by deltanedas** [comment]
yes but i believe firestacks ignore any kind of defence so if you light him on fire boom
also bombs 1 shot him !!!!!!

**Comment by HoidC** [comment]
> yes but i believe firestacks ignore any kind of defence so if you light him on fire boom
> also bombs 1 shot him !!!!!!

terminator shouldn’t be able to put armour on then because they can just get a bomb suit or RD suit and become omega ultra impossible to kill due to resistances 

maybe make their skin block the armour slot or something 

incinedary ammo being the only method to kill it after that means you’ll just get nukie armoury issue where all the counters are removed by smart players

0.1 resist on top of normal move speed is enough to be considered impregnable to rifles anyway considering how many full magazines it’d take to deskin them anyway

**Comment by deltanedas** [comment]
1. get beaker of chlorine
2. get beaker of flourine
3. chuck them at arnold
4. :trollface:

**Comment by HoidC** [comment]
> * get beaker of chlorine
> * get beaker of flourine
> * chuck them at arnold
> * :trollface:



> 1. get beaker of chlorine
> 2. get beaker of flourine
> 3. chuck them at arnold
> 4. :trollface:
> 
> anyway good news lasers will work because fire actually dont ignore the damage thing

so lasers deal full damage? cool lasers have a use now this truly is a utopian society

**Comment by deltanedas** [comment]
naked person not on manifest time to raid armoury for lasers :trollface:

File: Content.Server/GameTicking/Rules/TerminatorRuleSystem.cs
Comment (deltanedas) [reply Content.Server/GameTicking/Rules/TerminatorRuleSystem.cs:43]: so should i still do it given above will happen...

**Comment by HoidC** [comment]
> naked person not on manifest time to raid armoury for lasers :trollface:

can’t wait for security to start gunning down every naked person they see

File: Content.Server/GameTicking/Rules/TerminatorRuleSystem.cs
Comment (LankLTE) [reply Content.Server/GameTicking/Rules/TerminatorRuleSystem.cs:43]: That's what happens with events like initial infected, so it's probably fine

**Comment by Equivocateur** [comment]
> “Quick there is a naked person who isn’t registered they must be an assassin sent from the future to kill our head chef”

I've seen unregistered George Melons' go by hop and hos numerous times. Not an issue.

Synths dont breathe to my knowledge, but a troller with a nitrous oxide canister can easily disarm him.
Additionally arnold should get structural damage on fists so he could just terminate walls between him and the target.

**Comment by Equivocateur** [comment]
Same akshually applies to a CMO armed with chloral hydrate, t-800 got no real bloodstream

**Comment by deltanedas** [comment]
>  but a troller with a nitrous oxide canister can easily disarm him.

no breathe so no

> Additionally arnold should get structural damage on fists so he could just terminate walls between him and the target.
pro

**Comment by deltanedas** [comment]
> Same akshually applies to a CMO armed with chloral hydrate, t-800 got no real bloodstream

both stages immune to nocturine

**Comment by Equivocateur** [comment]
Does he get a pinpointer to his target so he wouldn't just shamble aimlessly around the station looking a cargo tech sleeping in a random locker in maints with an oxy tank - to not make an admin pissed that the player carrying out the smite can't do his job?

**Comment by Flareguy** [comment]
What's stopping the terminator from killing everyone? With increased damage resistances you're going to become very equipped to handle that kind of stuff.

**Comment by deltanedas** [comment]
> Does he get a pinpointer to his target so he wouldn't just shamble aimlessly around the station looking a cargo tech sleeping in a random locker in maints with an oxy tank - to not make an admin pissed that the player carrying out the smite can't do his job?

use a phonebook :trollface:

**Comment by deltanedas** [comment]
> What's stopping the terminator from killing everyone? With increased damage resistances you're going to become very equipped to do that kind of stuff.

chlorine and fluorine, incendiary shells

**Comment by Equivocateur** [comment]
> > What's stopping the terminator from killing everyone? With increased damage resistances you're going to become very equipped to do that kind of stuff.
> 
> chlorine and fluorine, incendiary shells

Unironic application of a welderbomb maybe?

**Comment by EmoGarbage404** [comment]
This needs a design document

**Comment by deltanedas** [comment]
the 6 lines of ??? was the document :trollface:

**Comment by MaloTV** [comment]
Does he get tc or any weapons on spawn or is he just a naked screaming man going rogue

**Comment by deltanedas** [comment]
watch the film bruh do i need to post a magnet

**Comment by JoeHammad1844** [comment]
are they space proof?

**Comment by JoeHammad1844** [comment]
also change their name to NT-800 :trollface:

**Comment by deltanedas** [comment]
practically spaceproof but itll eventually die if you stay in space forever

**Comment by mirrorcult** [comment]
pretty sure terminator and skynet are copyrighted

**Comment by ps3moira** [comment]
Hey, I was sent this PR when I made some sprites for a Terminator Mid-Round antag and turns out you already were coding it! Anyway, I have sprites for the terminator I'd like to suggest for the base sprites, and what it can look like in-game. My discord is ps3moira on the SS14 discord server.
Screenshot 202
![Screenshot 2023-09-08 194816
3-09-08 195424](https://github.com/space-wizards/space-station-14/assets/113228053/43513070-2a9b-4544-81e9-a4a70d8a5358)

**Comment by asperger-sind** [comment]
> discord is ps3moira on the SS14 discord server. 

delta doesnt use discord he's a schizo

**Comment by deltanedas** [comment]
just dump sprites :trollface:

> pretty sure terminator and skynet are copyrighted

as i said, i am not a lawyer

**Comment by mirrorcult** [comment]
not being a lawyer doesnt give you the ability to break copyright law

**Comment by deltanedas** [comment]
exterminator, model nt-800 :trollface:

**Comment by NancokPS2** [comment]
> not being a lawyer doesnt give you the ability to break copyright law

It really shouldn't be an issue.  
Referrinf to a robot as a terminator is not exactly copyrighted (unless we go as far as to actually name it T-800)  
  
Also xenos are a thing.

**Comment by deltanedas** [comment]
skynet -> susnet
cyberdyne -> cybersun in the future
t-800 parts -> nt-800 parts
terminator -> exterminator

tesco value now :trollface:

**Review by EmoGarbage404 — DISMISSED** [review]
I don't know if you misunderstood my initial comment, but this PR will *require* a design document. Our policy requires that large content features. especially antagonists, come with a design doc outlining intended features and functionality.

We have a template [here](https://hackmd.io/@ss14-design/Bk_ER9S4h) on the SS14 hackmd.

This PR isn't to be merged until that has been provided and approved.

**Comment by deltanedas** [comment]
@EmoGarbage404 pro done

**Comment by NancokPS2** [comment]
There's a couple of things that the doc should probably have:    

- To what extent can it act to fulfill it's objective? is it allowed to kill anyone in it's path to reach it's goal?   
- If it's only vulnerable to blunt damage, what are the main ways of fighting it? Who is realistically expected to go up against it and with what weapon?  
- How many can occur per round? What is the expected difficulty when compared to other antags and with which other antags is it compatible?  
- In terms of rules, to what extent can it collaborate with other antags?

**Comment by deltanedas** [comment]
> If it's only vulnerable to blunt damage, what are the main ways of fighting it? Who is realistically expected to go up against it and with what weapon?

thats endoskeleton and basically anyone since it cant hold guns just dogpile it

**Comment by deltanedas** [comment]
@EmoGarbage404 do you have anything to add wrt the design doc

**Comment by JoeHammad1844** [comment]
how does it heal? bruise packs? welders?

**Comment by deltanedas** [comment]
bruise packs then if you are exoskeleton someone else has to weld you :trollface:

**Comment by noverd** [comment]
терем-nator

**Comment by Froozigiusz** [comment]
Which naming convention is the correct one? t–800 as presented in the character screen screenshot, or NT–800 as suggested by the changelog? I'd like to know which one is the official one, and I suggest maybe sticking to one.

**Comment by deltanedas** [comment]
im pretty sure i changed it to say nt-800 ingame everywhere...

**Comment by ps3moira** [comment]
I've finished the sprites! 
image

I wanted to ask, would you add a function where terminators can turn on their eyes like flashlights? If so I can make glowing red eyes version of the sprite.

**Comment by KingOnBudget** [comment]
Where will it spawn?

**Comment by deltanedas** [comment]
random place in maints

**Comment by JoeHammad1844** [comment]
how is it determined? with a marker mappers have to place?

**Comment by deltanedas** [comment]
event creates a spawn point at a random place, ghost waits until nobody is watching to spawn

**Comment by ps3moira** [comment]
image
terminator

Here's the sprites for the EX-terminator.

**Comment by ps3moira** [comment]
image
image_2023-09-22_140500369_scaled_16x_pngcrushed

Here's the sprites sheet! you can credit ps3moira#9488

**Comment by deltanedas** [comment]
real

**Comment by RedNiloc** [comment]
God, I would love to see this get added to the game.

**Comment by noverd** [comment]
1984

**Comment by Emisse** [comment]
come to discord for free maintainer

**Comment by JoeHammad1844** [comment]
it would actually be pretty fun to be hunted down

**Comment by noverd** [comment]
died

**Comment by IProduceWidgets** [comment]
This is great. Was mulling over how to accomplish this for the last month or so, and I think you've pretty much ticked all the boxes I couldn't figure out.

**Comment by deltanedas** [comment]
exterminator 2: trial day

**Comment by crazybrain23** [comment]
If it's a kind of cyborg, will it have AI laws?

**Comment by deltanedas** [comment]
its not borg

**Comment by deltanedas** [comment]
@EmoGarbage404 will you look at the design doc :trollface:

**Comment by deltanedas** [comment]
design doc is merged someone remove DO NOT MERGE :trollface:

**Comment by revsys413** [comment]
if this gets merged soon after thief did i’m going to be so hyped

**Comment by Hmeister-fake** [comment]
Merge

**Comment by Emisse** [comment]
about to get demoted!!!

**Comment by mirrorcult** [comment]
dawg

**Comment by mirrorcult** [comment]
well its probably fine tbh

**Comment by deltanedas** [comment]
realops

**Comment by Hmeister-fake** [comment]
Literally buzzing, ss14 finally getting the stuff it needs. Next merge is going to be surgery (real)

**Pull request merged**
